piksi_multi_cpp/include/piksi_multi_cpp/receiver.h

```diff
@@ -0,0 +1,86 @@
+#ifndef PIKSI_MULTI_CPP_RECEIVER_H_
+#define PIKSI_MULTI_CPP_RECEIVER_H_
+
+#include <libsbp/sbp.h>
+#include <ros/ros.h>
+#include <memory>
+#include <string>
+#include <vector>
+#include "piksi_multi_cpp/callback.h"
+#include "piksi_multi_cpp/device.h"
+
+namespace piksi_multi_cpp {
+
+// This class offers a ROS interface for Piksi Multi receivers. It automatically
+// assigns a type to a device and offers ROS topics and services.
+class Receiver {
+ public:
+  /* The three types of receivers are
+
+  kBaseStationReceiver: The static base station sending out RTK corrections.
+
+  kPositionReceiver: The moving rover receiving RTK corrections from the
+  base station and broadcasting RTK GPS positions.
+
+  kAttitudeReceiver: The moving rover receiving RTK corrections from a moving
+  reference receiver and broadcasting the moving baseline (also referred to as
+  heading). */
+  enum Type {
+    kBaseStationReceiver = 0,
```

Comment:
Here there is also quite some ambiguity - we have an enum for this, and a class hierarchy. I would do away with the enum and use the class hierarchy only. Any other class using these types doesn't need to know which it is, otherwise it breaks the some separation-of-concern principles